src/common/resource/CResourceRef.cpp

```diff
@@ -17,7 +17,12 @@ CResourceRef::CResourceRef()
 CResourceRef::CResourceRef(CResourceLink* pLink) : m_pLink(pLink)
 {
     ASSERT(pLink);
-    pLink->AddRefInstance();
+    /*

```

Comment:
My doubt here is: what if, instead, one wants to check the number or loop through the instances, including the dupe items? This solution would cause the opposite issue.
An idea is to add a separate counter for strict instances and dupe instances, increased with AddRefInstanceStrict and AddRefInstanceDupe (valid only for items). In this case, FORINSTANCES needs to accept an additional parameter, 0 (default, with dupe item instances)/1 (strict, only direct instances).
I'd like to hear from other devs and contributors here!

Reply:
In any case, reading the issue #444 it looks like we need to decrease the strict ref counter or to avoid its increment when the ID changes

Reply:
I guess that because I haven't explained myself well. FORINSTANCES still working fine, this pLink is called when sphere read dupelist and every item added 1 point in instance, but when I just cancel this, FORINSTANCES and SERV.ITEMDEF.item.INSTANCES still working fine. 

How I tested:
I added i_bee_swarm 2 times.
Flip one of them,
I added one of the i_bee_swarm item from it's dupelist

I used FORINSTANCE loop on i_bee_swarm - returned 3
I used .show serv.itemdef.i_bee_swarm.instances - return 3

I haven't seen any issue on it but maybe I am wrong, I am just trying to get best result, and give best information I can give :)